Firmware-ESP/src/homeassistant.cpp

```diff
@@ -260,8 +411,31 @@ void removeMQTTButtonEvent(device_t device) {
 }
 
 void handleMQTTCommand(char* command) {
-	if (strncmp(command, "test_devices", 12) == 0)
-		sendTestButtonMsg();
+	if (strncmp(command, "test_devices", 12) == 0) {
+		sendTestButtonMsg(DEVICE_TYPE_ALL);
+	} 
+	else if (strncmp(command, "test_smoke", 10) == 0) { 
+		sendTestButtonMsg(DEVICE_TYPE_SMOKE);
+	} 
+	else if (strncmp(command, "test_co", 7) == 0) { 
+		sendTestButtonMsg(DEVICE_TYPE_CO);
+	} 
+	else if (strncmp(command, "silence_smoke", 13) == 0) {
+		sendSilenceButtonMsg(DEVICE_TYPE_SMOKE);
+	} 
+	else if (strncmp(command, "silence_co", 10) == 0) {
+		sendSilenceButtonMsg(DEVICE_TYPE_CO);
+	}
+	else if (strncmp(command, "emergency_smoke", 15) == 0) {
+		sendEmergencyButtonMsg(DEVICE_TYPE_SMOKE);
+	}
+	else if (strncmp(command, "emergency_co", 12) == 0) {
+		sendEmergencyButtonMsg(DEVICE_TYPE_CO);
+	}
```

Comment:
As discussed in #2, I'm not really a fan of having this functionality available, but looks like I'm the only one :smile:  Yes, it's disabled in the MQTT auto discovery, but I'd be happy if we could allow users to strip this out at build time (e.g. `#ifdef ENABLE_SILENCE_ALARM` and `#ifdef ENABLE_TRIGGER_ALARM`). Having this build flag default to enabled would be fine for me.

Reply:
Good idea! Added this in and defaulted both silencing and triggering alarms to be disabled by default.